plugins/out_http/http.c

```diff
@@ -136,7 +141,33 @@ static int http_post(struct flb_out_http *ctx,
     c->cb_ctx = ctx->ins->callback;
 
     /* Append headers */
-    if ((ctx->out_format == FLB_PACK_JSON_FORMAT_JSON) ||
+    if (headers) {
```

Comment:
please move this logic to a separate function. 

question: why does setting up headers exclude the other formats ?